
This patch makes bottom field first as default on interlaced content when DRM EGLdmabuf is used.
Also patch sets deinterlacer level acordingly to video profile settings.

diff --speed-large-files --no-dereference --minimal -Naur mythtv-d532d9950b967b3f5a4103b0de51111563ef0ca9/mythtv/libs/libmythtv/opengl/mythdrmprimeinterop.cpp mythtv-d532d9950b967b3f5a4103b0de51111563ef0ca9/mythtv/libs/libmythtv/opengl/mythdrmprimeinterop.cpp
--- mythtv-d532d9950b967b3f5a4103b0de51111563ef0ca9/mythtv/libs/libmythtv/opengl/mythdrmprimeinterop.cpp	2024-07-30 18:22:24.000000000 +0200
+++ mythtv-d532d9950b967b3f5a4103b0de51111563ef0ca9/mythtv/libs/libmythtv/opengl/mythdrmprimeinterop.cpp	2024-07-31 18:58:31.623340564 +0200
@@ -184,12 +184,12 @@
 
     // Is deinterlacing selected? Accept any value as RGB frames can only be deinterlaced here
     bool doublerate = false;
-    MythDeintType option = Frame->GetDoubleRateOption(DEINT_CPU | DEINT_SHADER | DEINT_DRIVER, DEINT_ALL);
-    if (option != DEINT_NONE)
+    MythDeintType deinterlacer = Frame->GetDoubleRateOption(DEINT_CPU | DEINT_SHADER | DEINT_DRIVER, DEINT_ALL);
+    if (deinterlacer != DEINT_NONE)
         doublerate = true;
     else
-        option = Frame->GetSingleRateOption(DEINT_CPU | DEINT_SHADER | DEINT_DRIVER, DEINT_ALL);
-    interlaced &= option != DEINT_NONE;
+        deinterlacer = Frame->GetSingleRateOption(DEINT_CPU | DEINT_SHADER | DEINT_DRIVER, DEINT_ALL);
+    interlaced &= deinterlacer != DEINT_NONE;
 
     // Clear redundant frame caches
     if (interlaced && !m_deinterlacing)
@@ -229,14 +229,19 @@
     if (m_deinterlacing)
     {
         result.clear();
-        Frame->m_deinterlaceInuse = DEINT_DRIVER | DEINT_BASIC;
+        if (deinterlacer == DEINT_HIGH)
+            Frame->m_deinterlaceInuse = DEINT_DRIVER | DEINT_HIGH;
+        else if (deinterlacer == DEINT_MEDIUM)
+            Frame->m_deinterlaceInuse = DEINT_DRIVER | DEINT_MEDIUM;
+        else
+            Frame->m_deinterlaceInuse = DEINT_DRIVER | DEINT_BASIC;
         Frame->m_deinterlaceInuse2x = doublerate;
         bool tff = Frame->m_interlacedReverse ? !Frame->m_topFieldFirst : Frame->m_topFieldFirst;
         int value {0};
         if (Scan == kScan_Interlaced)
-            value = tff ? 1 : 0;
-        else
             value = tff ? 0 : 1;
+        else
+            value = tff ? 1 : 0;
         result.emplace_back(m_openglTextures[id].at(value));
     }
 
